Fix CI by optimizing space

For fuzzer integration, our CI runs out of space, see: https://github.com/OpenZeppelin/polkadot-runtime-template/actions/runs/7888468429/job/21526068137

```
LLVM ERROR: IO failure on output stream: No space left on device
error: could not compile `polkadot-cli` (lib)
Error: Process completed with exit code 101.
```

I have identified the following potential optimizations:

- [x] we are building for both `debug` and `release` in our CI. We should only select one (the obvious choice is `release`)
- [x] we can strip symbols and debug information, those will not be useful in CI
- [x] we can use the `opt-level=s` flag to optimize the process for binary size

--- a/.github/workflows/cargo-tests.yml
+++ b/.github/workflows/cargo-tests.yml
@@ -26,6 +26,8 @@ env:
   CARGO_TERM_COLOR: always
   # Remove unnecessary WASM build artifacts
   WASM_BUILD_CLEAN_TARGET: 1
+  # stripping symbols and optimizing for binary size
+  RUSTFLAGS: -C strip=symbols -C opt-level=s
 
 jobs:
   clippy-fmt-test:
@@ -69,16 +71,16 @@ jobs:
         run: cargo fmt --all -- --check
 
       - name: Run tests
-        run: cargo test
+        run: cargo test --release
 
       - name: Check build
         run: cargo check --release
 
       - name: Check clippy
-        run: cargo clippy --locked --all-targets -- -D warnings
+        run: cargo clippy --release --locked --all-targets -- -D warnings
 
       - name: Check Documentation
-        run: cargo doc --locked --all --no-deps
+        run: cargo doc --release --locked --all --no-deps
         env:
           RUSTDOCFLAGS: "-D rustdoc::broken-intra-doc-links -D rustdoc::private_intra_doc_links"
 

